compiler/modules/front.syntax/preparse.bal

```diff
@@ -90,7 +90,40 @@ function preparseArrayTypeDesc(Tokenizer tok) returns boolean|err:Syntax {
         _ = check tok.expectIdentifier();
     }
     check tok.expect("[");
-
-    // SUBSET fixed length array types / tuples are not supported, is a td only if no token between `[` `]`
-    return tok.current() == "]";
+    while true {
```

Comment:
When there are multiple array dimensions such as `T[2][2]` we would have to iterate through expression right? Here after the first `[2]` since the next token is `[` we will continue the loop.